plugin.video.BlanketFort/default.py

```diff
@@ -0,0 +1,515 @@
+            except:
+                contextMenu = [('Add to BlanketFort Favorites','XBMC.Container.Update(%s?mode=5&name=%s&url=%s&iconimage=%s&fanart=%s)' %(sys.argv[0], urllib.quote_plus(name), urllib.quote_plus(url), urllib.quote_plus(iconimage), urllib.quote_plus(fanart)))]
+            liz.addContextMenuItems(contextMenu)
+        if not playlist is None:
+            playlist_name = name.split(') ')[1]
+            contextMenu_ = [('Play '+playlist_name+' PlayList','XBMC.RunPlugin(%s?mode=13&name=%s&playlist=%s)' %(sys.argv[0], urllib.quote_plus(playlist_name), urllib.quote_plus(str(playlist).replace(',','|'))))]
+            liz.addContextMenuItems(contextMenu_)
+        ok=xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=u,listitem=liz)
+        return ok
+
+
+  # Thanks to daschacka, an epg scraper for http://i.teleboy.ch/programm/station_select.php - http://forum.xbmc.org/showpost.php?p=936228&postcount=1076
+def getepg(link):
+        url=urllib.urlopen(link)
+        source=url.read()
+        url.close()
+        source2 = source.split("Jetzt")
+        source3 = source2[1].split('programm/detail.php?const_id=')
+        sourceuhrzeit = source3[1].split('<br /><a href="/')
+        nowtime = sourceuhrzeit[0][40:len(sourceuhrzeit[0])]
+        sourcetitle = source3[2].split("</a></p></div>")
+        nowtitle = sourcetitle[0][17:len(sourcetitle[0])]
+        nowtitle = nowtitle.replace("ö","oe")
+        nowtitle = nowtitle.replace("ä","ae")
+        nowtitle = nowtitle.replace("ü","ue")
+        return "  - "+nowtitle+" - "+nowtime
+
+
+xbmcplugin.setContent(int(sys.argv[1]), 'movies')
+try:
+    xbmcplugin.addSortMethod(int(sys.argv[1]), xbmcplugin.SORT_METHOD_UNSORTED)
+except:
+    pass
+try:
+    xbmcplugin.addSortMethod(int(sys.argv[1]), xbmcplugin.SORT_METHOD_LABEL)
+except:
+    pass
+try:
+    xbmcplugin.addSortMethod(int(sys.argv[1]), xbmcplugin.SORT_METHOD_DATE)
+except:
+    pass
+try:
+    xbmcplugin.addSortMethod(int(sys.argv[1]), xbmcplugin.SORT_METHOD_GENRE)
+except:
+    pass
+
+params=get_params()
+
+url=None
+name=None
+mode=None
+playlist=None
+
+try:
+    url=urllib.unquote_plus(params["url"]).decode('utf-8')
+except:
+    pass
+try:
+    name=urllib.unquote_plus(params["name"])
+except:
+    pass
+try:
+    iconimage=urllib.unquote_plus(params["iconimage"])
+except:
+    pass
+try:
+    fanart=urllib.unquote_plus(params["fanart"])
+except:
+    pass
+try:
+    mode=int(params["mode"])
+except:
+    pass
+try:
+    playlist=eval(urllib.unquote_plus(params["playlist"]).replace('|',','))
+except:
+    pass
+
+print "Mode: "+str(mode)
+if not url is None:
+    print "URL: "+str(url.encode('utf-8'))
+print "Name: "+str(name)
+
+
+if mode==None:
+    print "getData"
+    data = 'http://pastebin.com/raw/Cnm4FeBJ' 

```

Comment:
I need to check, but this may not be an acceptable source for items. Is there a website url available instead?